framework/sandstone.cpp

```diff
@@ -2685,6 +2706,121 @@ static void warn_deprecated_opt(const char *opt)
             program_invocation_name, opt);
 }
 
+
+static bool system_is_idle(float idle_threshold)
+{
+    #ifdef __linux__
+
+    FILE *loadavg;
+    float load_1, load_5, load_15;
+    int ret;
+
+    // Look at loadavg for hints whether the system is reasonably idle or not
+    // any error and we assume it's busy/under load for simplicity
+
+    loadavg = fopen("/proc/loadavg", "r");
+    if (!loadavg)
+        return false;
+
+    ret = fscanf(loadavg, "%f %f %f", &load_1, &load_5, &load_15);
+    fclose(loadavg);
+
+    if (ret != 3)
+       return false;
+
+    if (load_5 <= idle_threshold)
+       return true;
+    #else //__linux__
+    return true;//TO DO implement idle trigger for Windows
+    #endif
+
+    return false;
+}
+
+static void background_scan_init(void)
+{
+    MonotonicTimePoint now = MonotonicTimePoint::clock::now();
+
+    // init timestamps to 1day old - this quickstarts testing on first run
+    if ((sApp->background_scan.init_timestamp == true) && (sApp->service_background_scan == true)) {
+        for (auto i = 0; i < sApp->background_scan.number_of_timestamps; i++) {
+            MonotonicTimePoint timestamp_init_val = 
+                    (now - sApp->background_scan.day_delay_between_tests);
+            sApp->background_scan.timestamp[i] = timestamp_init_val;
+        }
```

Comment:
C++ hints (totally unnecessary change, but for learning). You can write this for loop using a ranged for:
```c++
    MonotonicTimePoint timestamp_init_val =
        (now - sApp->background_scan.day_delay_between_tests);
    for (MonotonicTimePoint &p : sApp->background_scan.timestamp)
        p = timestamp_init_val;
````

Or you can just use `std::array`'s own [`fill`](https://en.cppreference.com/w/cpp/container/array/fill) function:
```c++
    sApp->backgroun_scan.timestamp.fill(now - sApp->background_scan.day_delay_between_tests));
```

Sometimes I prefer writing the `for` loop myself too, when it's un-obvious what the <algorithm> function is doing or it looks uglier. Therefore, since it's about taste, I'll leave this up to you.